src/table/components/TableHeadWrapper.jsx

```diff
@@ -18,16 +19,21 @@ function TableHeadWrapper({
   translator,
   selectionsAPI,
   keyboard,
-  children,
 }) {
   const { columns, paginationNeeded } = tableData;
+  const setHeadRowHeight = useContextSelector(TableContext, (value) => value.setHeadRowHeight);
   const isFocusInHead = useContextSelector(TableContext, (value) => value.focusedCellCoord[0] === 0);
   const setFocusedCellCoord = useContextSelector(TableContext, (value) => value.setFocusedCellCoord);
   const headerStyle = useMemo(() => getHeaderStyle(layout, theme), [layout, theme]);
+  const headRowRef = useRef();
+
+  useEffect(() => {
+    setHeadRowHeight(headRowRef.current.clientHeight);
+  }, [headerStyle.fontSize]);
```

Comment:
I checked it in my branch with/without dynamic total height and it looks working, also tried in your branch didn't face similar issue! on which commit did you build it ? Is there any specific settings? (e.x fontszie).

Reply:
In any case it is good practice to include the variables that are used inside the useEffect. (We don't do it in some places in the repo, since we don't want to trigger updates for those "missing" variables)